pkg/kv/kvserver/batcheval/cmd_add_sstable.go

```diff
@@ -268,6 +268,21 @@ func EvalAddSSTable(
 	reply.RangeSpan = cArgs.EvalCtx.Desc().KeySpan().AsRawSpanWithNoLocals()
 	reply.AvailableBytes = cArgs.EvalCtx.GetMaxBytes() - cArgs.EvalCtx.GetMVCCStats().Total() - stats.Total()
 
+	if args.CheckForDataAbove {
+		existingIter := readWriter.NewMVCCIterator(
+			storage.MVCCKeyIterKind, // don't care if it is committed or not, just that it isn't empty.
+			storage.IterOptions{UpperBound: reply.RangeSpan.EndKey},
+		)
+		defer existingIter.Close()
+		existingIter.SeekGE(end)
```

Comment:
I think this requires taking out a read latch over this span. Is that going to be disruptive to other AddSSTable requests into the range?

Reply:
Ah, yeah, is an excellent point. I could get this key from AdminSplit instead and it already had a latch over the whole range. plus I only need this key when the range is full, which is when I'm going to split it, not every time I add to it. @nvanbenschoten I think this might suggest we revisit asking AdminSplit to reply with the first key in the new RHS (we're already looking for it for the stats choice too)

Reply:
That seems fine at a first glance, but I know @nvanbenschoten had some thoughts on this so I'll defer to him.